Chaincase.UI/ViewModels/ReceiveViewModel.cs

```diff
@@ -36,6 +47,45 @@ public void InitNextReceiveKey()
         public HdPubKey ReceivePubKey { get; set; }
 
         public string BitcoinUri => $"bitcoin:{Address}";
+        public string P2EPUri => $"bitcoin:{Address}?pj={_P2EPServer.PaymentEndpoint}";
+
+        public void GenerateP2EP(string password)
+        {
+            if (!_P2EPServer.HiddenServiceIsOn)
+            {
+                StartPayjoin(password);
+            }
+            P2EPAddress = _P2EPServer.PaymentEndpoint;
+        }
+
+        public void StartPayjoin(string password)
+        {
+            var cts = new CancellationToken();
+            _P2EPServer.StartAsync(cts);
+            _P2EPServer.Password = password;
```

Comment:
field may be better set from the constructor so the consumer doesn't need to know this bit of _P2EPServer nternals